### PR TITLE
Add inline documentation for `decode` functions

### PR DESCRIPTION
**Pull request by gfontenot**
This will need updating assuming #339 is accepted

### CODE REVIEW

File: Argo/Functions/decode.swift [hunk 1]
```diff
@@ -1,35 +1,232 @@
...
+  - returns: An `Optional<T>` value where `T` is `Decodable`
+*/
+public func decode<T: Decodable where T == T.DecodedType>(object: AnyObject) -> T? {
+  return decode(object).value
 }
 
-public func decodeWithRootKey<T: Decodable where T == T.DecodedType>(rootKey: String, _ object: AnyObject) -> [T]? {
-  return decodeWithRootKey(rootKey, object).value
+/**
+  Attempt to transform `AnyObject` into an `Array` of `Decodable` values and
```
Comment (tonyd256) [inline Argo/Functions/decode.swift]: `Optional`*

File: Argo/Functions/decode.swift [hunk 2]
```diff
@@ -1,35 +1,232 @@
...
+  return an `Optional`.
```
Comment (tonyd256) [inline Argo/Functions/decode.swift]: oh nevermind

File: Argo/Functions/decode.swift [hunk 3]
```diff
@@ -1,35 +1,232 @@
...
+  do {
+    let object = try NSJSONSerialization.JSONObjectWithData(data, options: nil)
+    let str: [String]? = decode(object)
+  } catch {
+    // handle error
+  }
+  ```
+
+  - parameter object: The `AnyObject` instance to attempt to decode
+  - returns: An `Optional<[T]>` value where `T` is `Decodable`
```
Comment (tonyd256) [inline Argo/Functions/decode.swift]: add extra line above

File: Argo/Functions/decode.swift [hunk 4]
```diff
@@ -1,35 +1,232 @@
...
+  do {
+    let object = try NSJSONSerialization.JSONObjectWithData(data, options: nil)
+    let str: String? = decode(object)
+  } catch {
+    // handle error
+  }
+  ```
+
+  - parameter object: The `AnyObject` instance to attempt to decode
+  - returns: An `Optional<T>` value where `T` is `Decodable`
```
Comment (tonyd256) [inline Argo/Functions/decode.swift]: add extra line above

File: Argo/Functions/decode.swift [hunk 5]
```diff
@@ -1,35 +1,232 @@
...
+  do {
+    let object = try NSJSONSerialization.JSONObjectWithData(data, options: nil)
+    let str: Decoded<[String]> = decode(object)
+  } catch {
+    // handle error
+  }
+  ```
+
+  - parameter object: The `AnyObject` instance to attempt to decode
+  - returns: A `Decoded<[T]>` value where `T` is `Decodable`
```
Comment (tonyd256) [inline Argo/Functions/decode.swift]: add extra line above

File: Argo/Functions/decode.swift [hunk 6]
```diff
@@ -1,35 +1,232 @@
...
+  do {
+    let object = try NSJSONSerialization.JSONObjectWithData(data, options: nil)
+    let str: Decoded<String> = decode(object)
+  } catch {
+    // handle error
+  }
+  ```
+
+  - parameter object: The `AnyObject` instance to attempt to decode
+  - returns: A `Decoded<T>` value where `T` is `Decodable`
```
Comment (tonyd256) [inline Argo/Functions/decode.swift]: add extra line above

File: Argo/Functions/decode.swift [hunk 7]
```diff
@@ -1,35 +1,232 @@
...
+
+/**
+  Attempt to transform `AnyObject` into an `Array` of `Decodable` value using a
+  specified root key and return an `Optional`
+
+  This function expects the object to be a dictionary of the type `[String: JSON]`.
+
+  This function attpemts to extract the embedded JSON object from the
+  dictionary at the specified key and transform it into an `Array` of
+  `Decodable` values, returnin an `Optional`. This works based on the type you
```
Comment (tonyd256) [inline Argo/Functions/decode.swift]: returning*

**Comment by tonyd256** [comment]
:+1:

**Comment by klaaspieter** [comment]
:+1: 

I'm slightly annoyed at the repetition. This entire suite of methods is essentially the same method with slightly different types. So by extension the documentation is essentially the same with slightly different types. When things inevitably change (#339) the change author has to remember to change all the affected documentation blurbs.

I'm unsure as to how and if to fix this, but I had a discussion some time ago with a friend about how code and documentation shouldn't be in the same file, and I kind of get what he meant now :stuck_out_tongue_winking_eye:

File: Argo/Functions/decode.swift (hunk 1 above)
Comment (gfontenot) [inline Argo/Functions/decode.swift]: This is correct, I say that we're returning an optional on the next line.

**Pull request merged**